File: backups/capacitor-sora-editor-1.3.13/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/snippet/variable/CompositeSnippetVariableResolver.java

package io.github.abc15018045126.sora.widget.snippet.variable;

import androidx.annotation.NonNull;

import java.util.HashMap;
import java.util.Map;
import java.util.Objects;

public final class CompositeSnippetVariableResolver implements ISnippetVariableResolver {

    private final Map<String, ISnippetVariableResolver> resolverMap;

    public CompositeSnippetVariableResolver() {
        resolverMap = new HashMap<>();
    }

    public void addResolver(@NonNull ISnippetVariableResolver resolver) {
        if (resolver instanceof CompositeSnippetVariableResolver) {
            throw new IllegalArgumentException();
        }
        Objects.requireNonNull(resolver, "resolver must not be null");
        for (var name : resolver.getResolvableNames()) {
            resolverMap.put(name, resolver);
        }
    }

    public void removeResolver(@NonNull ISnippetVariableResolver resolver) {
        for (var name : resolver.getResolvableNames()) {
            if (resolverMap.get(name) == resolver) {
                resolverMap.remove(name);
            }
        }
    }

    @NonNull
    @Override
    public String[] getResolvableNames() {
        return new String[0];
    }

    public boolean canResolve(@NonNull String name) {
        return resolverMap.containsKey(name);
    }

    @NonNull
    @Override
    public String resolve(@NonNull String name) {
        var resolver = resolverMap.get(name);
        if (resolver != null) {
            return resolver.resolve(name);
        }
        return "";
    }

}

File: capacitor-sora-editor/android/sora-editor/editor/src/main/java/io/github/abc15018045126/sora/widget/snippet/variable/CompositeSnippetVariableResolver.kt
package io.github.abc15018045126.sora.widget.snippet.variable

class CompositeSnippetVariableResolver : ISnippetVariableResolver {

    private val resolverMap = HashMap<String, ISnippetVariableResolver>()

    fun addResolver(resolver: ISnippetVariableResolver) {
        if (resolver is CompositeSnippetVariableResolver) {
            throw IllegalArgumentException("Cannot add a CompositeSnippetVariableResolver to another one")
        }
        for (name in resolver.getResolvableNames()) {
            resolverMap[name] = resolver
        }
    }

    fun removeResolver(resolver: ISnippetVariableResolver) {
        for (name in resolver.getResolvableNames()) {
            if (resolverMap[name] === resolver) {
                resolverMap.remove(name)
            }
        }
    }

    override fun getResolvableNames(): Array<String> {
        return emptyArray()
    }

    fun canResolve(name: String): Boolean {
        return resolverMap.containsKey(name)
    }

    override fun resolve(name: String): String {
        return resolverMap[name]?.resolve(name) ?: ""
    }
}
